### PR TITLE
cephadm: Update error message on host failure

### PR DESCRIPTION
**Pull request by Devp00l**
Now the error message explains how to add the ssh key to the not
reachable host

Fixes: https://tracker.ceph.com/issues/46551
Signed-off-by: Stephan Müller <smueller@suse.com>

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/cephadm/troubleshooting.rst [hunk 1]
```diff
@@ -147,8 +158,15 @@ Verifying that the Public Key is Listed in the authorized_keys file
...
 
-     [root@mon1 ~]# cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_pub > key.pub
-     [root@mon1 ~]# grep "`cat key.pub`"  /root/.ssh/authorized_keys
+     [root@mon1 ~]# cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_pub > ~/cephadm_ssh_key.pub
+
+or::
+
+     [root@mon1 ~]# ceph cephadm get-pub-key > ~/cephadm_ssh_key.pub
+
+than::
```
Comment (mgfritch) [inline doc/cephadm/troubleshooting.rst:167]:
```suggestion
then::
```

File: doc/cephadm/troubleshooting.rst [hunk 2]
```diff
@@ -147,8 +158,15 @@ Verifying that the Public Key is Listed in the authorized_keys file
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 To verify that the public key is in the authorized_keys file, run the following commands::
 
-     [root@mon1 ~]# cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_pub > key.pub
-     [root@mon1 ~]# grep "`cat key.pub`"  /root/.ssh/authorized_keys
+     [root@mon1 ~]# cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_pub > ~/cephadm_ssh_key.pub
+
+or::
+
```
Comment (mgfritch) [inline doc/cephadm/troubleshooting.rst:164]:
```suggestion
```

remove this rather than list two commands that are essentially the same?

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-07-30_07:40:52-rados:cephadm-wip-swagner-testing-2020-07-29-1507-distro-basic-smithi/

File: doc/cephadm/troubleshooting.rst (hunk 2 above)
Comment (Devp00l) [reply doc/cephadm/troubleshooting.rst]: Removed it

File: doc/cephadm/troubleshooting.rst (hunk 1 above)
Comment (Devp00l) [reply doc/cephadm/troubleshooting.rst]: Fixed by removing the other option to retrieve the key, as no separation was longer needed.

File: doc/cephadm/troubleshooting.rst [hunk 3]
```diff
@@ -147,8 +158,8 @@ Verifying that the Public Key is Listed in the authorized_keys file
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 To verify that the public key is in the authorized_keys file, run the following commands::
 
-     [root@mon1 ~]# cephadm shell -- ceph config-key get mgr/cephadm/ssh_identity_pub > key.pub
-     [root@mon1 ~]# grep "`cat key.pub`"  /root/.ssh/authorized_keys
+     [root@mon1 ~]# ceph cephadm get-pub-key > ~/cephadm_ssh_key.pub
```
Comment (sebastian-philipp) [inline doc/cephadm/troubleshooting.rst:161]: no sure about this one. for ceph-salt: sure that would be better. but for upstream, `cephadm shell --` sounds better to me

Comment (Devp00l) [reply doc/cephadm/troubleshooting.rst]: Thanks for spotting this, changed it.

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
jenkins render docs

File: doc/cephadm/adoption.rst [hunk 4]
```diff
@@ -75,11 +75,11 @@ Adoption process
 #. Generate an SSH key::
 
      # ceph cephadm generate-key
-     # ceph cephadm get-pub-key > ceph.pub
+     # ceph cephadm get-pub-key > ~/cephadm_ssh_key.pub
```
Comment (sebastian-philipp) [inline doc/cephadm/adoption.rst:78]: Hm we already have `ceph.pub` hardcoded: https://github.com/ceph/ceph/search?q=%22ceph.pub%22

Hm. I have the fear that we might irrigate users if we suddenly have different names for the same thing.

File: doc/cephadm/troubleshooting.rst [hunk 5]
```diff
@@ -110,34 +110,45 @@ ssh errors
...
+  Please make sure that the host is reachable and accepts connections using the cephadm SSH key
+
+  To add the cephadm SSH key to the host:
+  > ceph cephadm get-pub-key > ~/cephadm_ssh_key.pub
+  > ssh-copy-id -f -i ~/cephadm_ssh_key.pub root@10.10.1.2
+
+  To check that the host is reachable:
+  > ceph cephadm get-ssh-config > ssh_config
+  > ceph config-key get mgr/cephadm/ssh_identity_key > ~/cephadm_private_key
+  > ssh -F ssh_config -i ~/cephadm_private_key root@10.10.1.2
```
Comment (sebastian-philipp) [inline doc/cephadm/troubleshooting.rst:126]: https://docs.ceph.com/ceph-prs/36330/cephadm/troubleshooting/#ssh-errors

this looks a bit odd and unreadable. Are you sure this is what users get?

File: doc/cephadm/troubleshooting.rst [hunk 6]
```diff
@@ -110,34 +110,45 @@ ssh errors
...
+
      [root@mon1 ~]# cephadm shell -- ceph cephadm get-ssh-config > config
 
 3. Verify that we can connect to the host::
-    
-     [root@mon1 ~]# ssh -F config -i key root@mon1
+
+     [root@mon1 ~]# ssh -F config -i ~/cephadm_private_key root@mon1
 
 4. There is a limitation right now: the ssh user is always `root`.
```
Comment (sebastian-philipp) [inline doc/cephadm/troubleshooting.rst:153]: cc  @Daniel-Pivonka

Comment (Daniel-Pivonka) [reply doc/cephadm/troubleshooting.rst:153]: i must have missed that in #35606

Comment (Daniel-Pivonka) [reply doc/cephadm/troubleshooting.rst:153]: #36706

**Comment by Devp00l** [comment]
Addressed all comments

**Comment by sebastian-philipp** [comment]
green: https://pulpito.ceph.com/swagner-2020-08-25_16:35:43-rados:cephadm-wip-swagner-testing-2020-08-25-1535-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
green: https://pulpito.ceph.com/swagner-2020-08-27_07:38:41-rados:cephadm-wip-swagner2-testing-2020-08-26-1329-distro-basic-smithi/

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**